src/protocol/sort/apply_sort/shuffle.rs

```diff
@@ -26,6 +26,48 @@ pub trait Resharable<F: Field>: Sized {
         C: Context<F, Share = <Self as Resharable<F>>::Share> + Send;
 }
 
+pub struct InnerVectorElementStep(usize);
+
+impl crate::protocol::Substep for InnerVectorElementStep {}
+
+impl AsRef<str> for InnerVectorElementStep {
+    fn as_ref(&self) -> &str {
+        const VEC_ELEM: [&str; 64] = [
+            "elem0", "elem1", "elem2", "elem3", "elem4", "elem5", "elem6", "elem7", "elem8",
+            "elem9", "elem10", "elem11", "elem12", "elem13", "elem14", "elem15", "elem16",
+            "elem17", "elem18", "elem19", "elem20", "elem21", "elem22", "elem23", "elem24",
+            "elem25", "elem26", "elem27", "elem28", "elem29", "elem30", "elem31", "elem32",
+            "elem33", "elem34", "elem35", "elem36", "elem37", "elem38", "elem39", "elem40",
+            "elem41", "elem42", "elem43", "elem44", "elem45", "elem46", "elem47", "elem48",
+            "elem49", "elem50", "elem51", "elem52", "elem53", "elem54", "elem55", "elem56",
+            "elem57", "elem58", "elem59", "elem60", "elem61", "elem62", "elem63",
+        ];
+        VEC_ELEM[self.0]
+    }
+}
+
+impl From<usize> for InnerVectorElementStep {
+    fn from(v: usize) -> Self {
+        Self(v)
+    }
+}
+
+#[async_trait]
+impl<F: Field> Resharable<F> for Vec<Replicated<F>> {
+    type Share = Replicated<F>;
+
+    async fn reshare<C>(&self, ctx: C, record_id: RecordId, to_helper: Role) -> Result<Self, Error>
+    where
+        C: Context<F, Share = <Self as Resharable<F>>::Share> + Send,
+    {
+        try_join_all(self.iter().enumerate().map(|(i, x)| {
+            let c = ctx.narrow(&InnerVectorElementStep::from(i));
```

Comment:
So we can only reshare a `Vec<>` with a length of 64 or less?  That is probably nto going to scale.

This seems like a good case for ignoring the record ID.  Or at least asserting that it is zero before ignoring it.  Alternatively, use the value provided as a starting value.

Reply:
The use-case for this is to sort the match-keys. Each match key is a [Replicated<F>; 40]. And we need to use RecordId to represent the row. So each row will have 40 bits within in. So I don't think either of the approaches you suggested will work.

Reply:
Oh, right.  That's a bit annoying, but I guess this has to be the way.

Reply:
Can you add a comment to the effect that this is for a vector of bit-decomposed values?